test/configs/websockets.json

```diff
@@ -0,0 +1,48 @@
+{
+  "@context": "https://linkedsoftwaredependencies.org/bundles/npm/@solid/community-server/^1.0.0/components/context.jsonld",
+  "import": [
+    "files-scs:config/presets/http.json",
+    "files-scs:config/presets/ldp/credentials-extractor.json",
+    "files-scs:config/presets/ldp/metadata-handler.json",
+    "files-scs:config/presets/ldp/operation-handler.json",
+    "files-scs:config/presets/ldp/permissions-extractor.json",
+    "files-scs:config/presets/ldp/response-writer.json",
+    "files-scs:config/presets/ldp/request-parser.json",
+    "files-scs:config/presets/ldp/websockets.json",
+    "files-scs:config/presets/representation-conversion.json",
+    "files-scs:config/presets/storage/backend/storage-memory.json",
+    "files-scs:config/presets/storage/routing/no-routing.json",
+    "files-scs:config/presets/storage-wrapper.json",
+    "files-scs:config/presets/cli-params.json"
+  ],
+  "@graph": [
+    {
+      "@id": "urn:solid-server:default:HttpHandler",
+      "@type": "AuthenticatedLdpHandler",
+      "AuthenticatedLdpHandler:_args_requestParser": {
+        "@id": "urn:solid-server:default:RequestParser"
+      },
+      "AuthenticatedLdpHandler:_args_credentialsExtractor": {
+        "@id": "urn:solid-server:default:CredentialsExtractor"
+      },
+      "AuthenticatedLdpHandler:_args_permissionsExtractor": {
+        "@id": "urn:solid-server:default:PermissionsExtractor"
+      },
+      "AuthenticatedLdpHandler:_args_authorizer": {
+        "@type": "AllowEverythingAuthorizer"
```

Comment:
I see a test config is made specifically for this line. The thing I'm wondering is how we exactly want to handle these test configs. Now it is saying that it specifically wants this authorizer, but for all the other things the defaults from the non-test config are ok. Depending on how our configs evolve and which integration tests we have this may or may not cause problems. For example, now the default credentials extractor is fine, but what when that one changes to an actual implementation?

Either we keep depending on the default configs, and change the test configs accordingly when something changes in the defaults (which could be a lot of changes if there are a lot of test configs), or we have a completely different set of test configs, but then we run the risk of our integration tests not testing what we are actually exporting with the default configs.

Not sure which one would be the best. Our current non-config integration tests correspond to the second one though.

Reply:
I purposely settled on this one.

> but then we run the risk of our integration tests not testing what we are actually exporting with the default configs

Exactly.

> now the default credentials extractor is fine, but what when that one changes to an actual implementation?

If it breaks, we want to know. Maybe it is predictable breakage or breakage by design, but maybe it is not.
In the former case, we fix the main config. In the second case, we adjust the test config.

Prematurely adjusting the test config makes it more likely that errors in the main config go undetected.
I also expect our test config to grow with a couple of reusable subconfigs for testing purposes.